src/pages/Questionnaire.tsx

```diff
@@ -82,8 +82,12 @@ interface QuesProps {
 export function Questionnaire() {
   const { questionId } = useParams<{ questionId: string }>();
   const navigate = useNavigate();
-  const [currentQuestion, setCurrentQuestion] = useState<number>(
-    questionId ? parseInt(questionId) : -1
+  const [currentQuestion, setCurrentQuestion] = useState<number>(-1);
+  const [responses, setResponses] = useState<Record<number, number | number[]>>(
+    () => {
+      const storedResponses = localStorage.getItem("questionnaire");
```

Comment:
you can fetch this item outside of the state.
and then do this:
` const [responses, setResponses] = useState(storedResponses ? JSON.parse(storedResponses) : {});`